### PR TITLE
Patch package #1

### PR DESCRIPTION
**Pull request by brendtumi**

### CODE REVIEW

File: src/components/Slider.vue [hunk 1]
```diff
@@ -47,25 +49,15 @@ export default {
...
-
-.slider::-webkit-slider-thumb {
-  -webkit-appearance: none;
-  appearance: none;
-  width: 25px;
-  height: 25px;
-  background: #4caf50;
-  cursor: pointer;
+.nvw-slider__scroll::-webkit-slider-thumb {
+  @include transition;
```
Comment (yerol) [inline src/components/Slider.vue]: You don't need mixin here. 

```
&__scroll {
    &::-webkit-slider-thumb,
    &::-moz-range-thumb {
        // transition code
    }
}
```

File: src/components/Span.vue [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    </span>
+</template>
+
+<script>
+export default {
+  name: 'Span',
+  props: {
+    text: String,
+  },
+  created() {},
```
Comment (yerol) [inline src/components/Span.vue]: remove unused created()

File: src/components/Switcher.vue [hunk 3]
```diff
@@ -1,23 +1,32 @@
...
+  props: {
+    checked: {
+      type: Boolean,
+      default: false,
+    },
+  },
   methods: {
     updateValue: function() {
-      this.$emit('input', this.$refs.switcher.checked);
+      this.$emit('input', this.$refs.switch.checked);
```
Comment (yerol) [inline src/components/Switcher.vue]: emitting input event is not needed. remove

File: src/components/Img.vue [hunk 4]
```diff
@@ -1,18 +1,53 @@
...
+      return {
+        'nvw-img': this.stretch === 'none' ? true : false,
+        'nvw-img nvw-img--fill': this.stretch === 'fill' ? true : false,
+        'nvw-img nvw-img--aspect-fill': this.stretch === 'aspectFill' ? true : false,
+        'nvw-img nvw-img--aspect-fit': this.stretch === 'aspectFit' ? true : false,
+      };
+    },
+  },
+  methods: {
+    convertPxStyle: function(value) {
```
[the remark this answers is not in the document]
Comment (mylmz10) [reply src/components/Img.vue]: Added helper function #8 PR

File: src/layouts/StackLayout.vue [hunk 5]
```diff
@@ -1,18 +1,26 @@
...
 </template>
 
 <script>
 export default {
   name: 'StackLayout',
-  props: ['tap', 'text'],
+  props: ['tap', 'text', 'col', 'row'],
+  computed: {
+    colNumber: function() {
+      return parseInt(this.col) + 1;
```
Comment (yerol) [inline src/layouts/StackLayout.vue]: parseInt() will throw if not number. Put it in try catch please

File: src/layouts/WrapLayout.vue [hunk 6]
```diff
@@ -1,30 +1,23 @@
...
-    return {
-      style: '',
-    };
-  },
-  mounted() {
-    this.style = `flex-direction: ${this.orientation == 'horizontal' ? 'row' : 'column'};`;
+  props: {
+    orientation: String,
+    // itemWidth and itemHeight will be set using directives for now.
+    itemWidth: Number,
```
Comment (yerol) [inline src/layouts/WrapLayout.vue]: on nativescript-vue docs it says itemWidth and itemHeight but the examples use width and height. let's drop these from this component and use the common directive to process width and heoght for each children

File: src/layouts/WrapLayout.vue [hunk 7]
```diff
@@ -1,30 +1,23 @@
...
+    itemHeight: Number,
   },
 };
 </script>
 
 <style lang="scss" scoped>
-.container {
+.nvw-wrap-layout {
   width: 50%;
```
Comment (yerol) [inline src/layouts/WrapLayout.vue]: why 50% width?

File: src/components/Slider.vue [hunk 8]
```diff
@@ -1,71 +1,59 @@
...
             ref="slider"
-            class="slider"
+            class="nvw-slider__scroll"
             type="range"
             :min="minValue"
             :max="maxValue"
             :value="value"
-            @input="updateValue()"
-            @change="onValueChange($event)"
+            @change="valueChange ? valueChange($event, value) : null"
```
Comment (yerol) [inline src/components/Slider.vue]: $event should always be the last argument

Comment (Ulubayam) [reply src/components/Slider.vue]: Can you review again?

File: src/components/Img.vue [hunk 9]
```diff
@@ -1,18 +1,44 @@
...
+    height: [String, Number],
+  },
   data() {
     return {
       imageSrc: `${this.src.replace('~', '')}`,
     };
   },
+  computed: {
+    stretchClass: function() {
+      return `nvw-img nvw-img${this.stretch !== 'none' ? '--' + camelCaseToDash(this.stretch) : ''}`;
```
Comment (yerol) [inline src/components/Img.vue]: When stretch is none it'll output "nvw-img nvw-img"

File: src/components/Switcher.vue [hunk 10]
```diff
@@ -27,71 +36,78 @@ export default {
...
-  top: 50%;
-  left: 50%;
-  font-size: 10px;
-  font-family: Verdana, sans-serif;
-}
+  & input:checked + .nvw-switch__slider:after {
+    content: 'ON';
+  }
+
+  & .nvw-switch__slider {
```
Comment (yerol) [inline src/components/Switcher.vue]: this should be "&__slider"

File: src/components/Switcher.vue [hunk 11]
```diff
@@ -27,71 +36,78 @@ export default {
...
+    position: absolute;
+    cursor: pointer;
+    @include pos();
+    background-color: #ca2222;
+    -webkit-transition: 0.5s;
+    transition: 0.4s;
+    border-radius: 34px;
+  }
+
+  & .nvw-switch__slider:before {
```
Comment (yerol) [inline src/components/Switcher.vue]: this should go inside &__slider with &:before

```
.nvw-switch {
    &__slider {
        &:before {}
    }
}
```

File: src/directives/CommonDirective.js [hunk 12]
```diff
@@ -0,0 +1,63 @@
+const CommonDirective = {
+  bind(el, binding, vnode) {
+    const { context } = vnode;
+    const { $attrs } = context;
+    // backgroundColor, width, height, top, left, flexShrink, flexGrow, order.
+    if (Object.keys($attrs).length) {
+      let cssString = '';
+      cssString = $attrs.backgroundColor
+        ? cssString
```
Comment (yerol) [inline src/directives/CommonDirective.js]: es6 templates are perfect for this. so many concats here.

`cssString += $attrs.backgroundColor ? `background-color:${$attrs.backgroundColor};` : '';`

File: src/helpers/lodash.debounce.js [hunk 13]
```diff
@@ -0,0 +1,374 @@
+/**
```
Comment (yerol) [inline src/helpers/lodash.debounce.js]: remove this file

File: src/layouts/AbsoluteLayout.vue [hunk 14]
```diff
@@ -1,23 +1,63 @@
...
+      const { propsData } = vnode.componentOptions.propsData;
+
+      for (const prop in propsData) {
+        if (!propsData.hasOwnProperty(prop)) continue;
+
+        switch (prop) {
+          case 'backgroundColor':
+            vnode.elm.style.backgroundColor = propsData[prop];
+            break;
+          case 'top':
```
Comment (yerol) [inline src/layouts/AbsoluteLayout.vue]: @ozercevikaslan we don't need these props anymore do we? please replace with common directive

File: src/layouts/FlexboxLayout.vue [hunk 15]
```diff
@@ -1,31 +1,77 @@
...
   width: 50%;
   height: 500px;
   background-color: crimson;
-  display: inline-flex;
+  display: flex;
+}
+
+.nvw-flexbox-layout > {
+  // Deafult stretch.
+  [alignSelf=''] or [alignSelf='stretch'] {
```
Comment (yerol) [inline src/layouts/FlexboxLayout.vue]: Can someone explain what "or" means in css? :)

File: src/layouts/WrapLayout.vue (hunk 6 above)
Comment (yerol) [reply src/layouts/WrapLayout.vue]: remove itemWidth and itemHeight props

File: src/components/Slider.vue (hunk 8 above)
Comment (yerol) [reply src/components/Slider.vue]: @Ulubayam it looks exactly the same to me. we have discussed this with @ozercevikaslan couple of hours back. Please use the same logic as in textfield for this. He will show you. Thanks

File: src/layouts/FlexboxLayout.vue (hunk 15 above)
Comment (ozercevikaslan) [reply src/layouts/FlexboxLayout.vue]: It evaluates the given statements as || operator does. I found it in the docs.

File: src/layouts/AbsoluteLayout.vue (hunk 14 above)
Comment (ozercevikaslan) [reply src/layouts/AbsoluteLayout.vue]: This comment will be a general reply to the issues in the components we did not consider to develop in this sprint. These components have not been updated yet in purpose not to lose time.

**Pull request merged**